uvicorn/config.py

```diff
@@ -385,6 +388,10 @@ def configure_logging(self) -> None:
                     self.log_config["formatters"]["access"][
                         "use_colors"
                     ] = self.use_colors
+                if self.access_log_format:
+                    self.log_config["formatters"]["access"][
+                        "fmt"
+                    ] = "%(levelprefix)s %(message)s"
```

Comment:
Shouldn't this be like:

```py
if self.access_log_format:
    self.log_config["formatters"]["access"][
        "fmt"
    ] = self.access_log_format

```

Reply:
Not quite: in Python logging does it happens in two stages. e.g. for `logger.info('message %(hello)s %(world)s', {'hello': 1, 'world': 2})`: 

- first `LogRecord.getMessage` will format the message with the args, i.e. expand `hello` and `world` and store it as `LogRecord.message`
```python
class LogRecord:
    def getMessage(self):
        msg = str(self.msg)
        if self.args:
            msg = msg % self.args
        return msg
``` 
- then LogRecord fields are formatted using the corresponding Formatter's `fmt`, that would expand `%(message)s` that contains `message 1 2` pre-formatted in the previous step
```python
class PercentStyle:
    def _format(self, record):
        return self._fmt % record.__dict__

# and

class Formatter:
    def formatMessage(self, record):
        return self._style.format(record)
```

So right now uvicorn's log formatting and gunicorn access log formatting happen in different stages of processing. They can be made to both work on `Formatter` level via a `LoggingAdapter` of sort, but we'll have to sort out some conflicts between the two log formats, `--access-log-format` coming from gunicorn and uvicorn's own `LOGGING_CONFIG`.